firefox-ios/Client/Frontend/Browser/TermsOfUse/TermsOfUseLinkViewController.swift

```diff
@@ -0,0 +1,117 @@
+    ) {
+        self.url = url
+        self.windowUUID = windowUUID
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+
+        let config = WKWebViewConfiguration()
+        config.setURLSchemeHandler(InternalSchemeHandler(shouldUseOldErrorPage: true), forURLScheme: InternalURL.scheme)
+        self.webView = WKWebView(frame: .zero, configuration: config)
+
+        super.init(nibName: nil, bundle: nil)
+
+        self.webView.navigationDelegate = self
+        self.webView.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        listenForThemeChange(view)
+        setupHeader()
+        setupWebView()
+        applyTheme()
+    }
+
+    private func setupHeader() {
+        let header = UIView()
+        header.backgroundColor = currentTheme().colors.layer1
+        header.translatesAutoresizingMaskIntoConstraints = false
+
+        let backButton = UIButton(type: .system)
+        backButton.setImage(UIImage(systemName: "chevron.backward"), for: .normal)
+        backButton.setTitle("Back", for: .normal)
+        backButton.tintColor = currentTheme().colors.actionPrimary
+        backButton.addTarget(self, action: #selector(closeTapped), for: .touchUpInside)
+        backButton.translatesAutoresizingMaskIntoConstraints = false
+
+        view.addSubview(header)
+        header.addSubview(backButton)
+
+        NSLayoutConstraint.activate([
+            header.topAnchor.constraint(equalTo: view.topAnchor),
+            header.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            header.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            header.heightAnchor.constraint(equalToConstant: UX.headerHeight),
+
+            backButton.leadingAnchor.constraint(equalTo: header.leadingAnchor, constant: UX.backButtonLeading),
+            backButton.centerYAnchor.constraint(equalTo: header.centerYAnchor)
+        ])
+    }
+
+    private func setupWebView() {
+        view.addSubview(webView)
+
+        NSLayoutConstraint.activate([
+            webView.topAnchor.constraint(equalTo: view.topAnchor, constant: UX.headerHeight),
+            webView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            webView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            webView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+
+        webView.load(URLRequest(url: url))
+    }
+
+    func applyTheme() {
+        view.backgroundColor = currentTheme().colors.layer1
+    }
+
+    @objc private func closeTapped() {
+        dismiss(animated: true)
+    }
+
+    private func currentTheme() -> Theme {
+        themeManager.getCurrentTheme(for: currentWindowUUID)
+    }
+}
+
+extension TermsOfUseLinkViewController: WKNavigationDelegate {
```

Comment:
We try not to use extensions for delegates. Instead move the protocol conformance to the class and use  // MARK to show where the delegate conformance is.